### ISSUE TITLE
Inconsistencies in where/how response models are shown (v2.1.2-M2)

### ISSUE THREAD

**Issue by xiehan** [swagger-api/swagger-ui#1265]
This is a continuation of #1147 - most of the original issues have been resolved but now I'm seeing some new, interesting issues. All of these were tested with v2.1.2-M2 in Chrome Canary. Our full Swagger doc is at http://api.npr.org/documentation/beryllium/api-docs which follows the v1.2 Swagger spec.

For this next one, part of my confusion, I think, relates to the fact that I don't know what the desired end result is -- where you guys intend for these to be shown on the page -- but either way, what's happening right now is very inconsistent.

In terms of schema, we pretty much always follow the same pattern:

``` javascript
            "path": "/identity/v2/following",
            "operations": [
                {
                    "method": "POST",
                    "summary": "Update the following status of the logged-in user for a particular aggregation",
                    "nickname": "updateFollowingStatus",
                    "type": "UserDocument", // <----------------
                    "parameters": [
                        ...
                    ],
                    "responseMessages": [
                        {
                            "code": 201,
                            "message": "Successfully updated the user's affiliation information",
                            "responseModel": "UserDocument" // <----------------
                        },
                        ... // error responses go below here
```

If you note those commented two arrows, you'll see that we essentially declare the response model for successful responses in two places.

In v2.1.8-M1, this resulted (consistently) in the following visual output:

![screen shot 2015-05-11 at 1 07 27 pm](https://cloud.githubusercontent.com/assets/855115/7570528/dd10bc0c-f7de-11e4-8fc0-cdbc08f46c8f.png)
![screen shot 2015-05-11 at 1 08 05 pm](https://cloud.githubusercontent.com/assets/855115/7570531/e05aac38-f7de-11e4-9aa7-4da2502d99ee.png)

It is perhaps a bit redundant, but honestly, it works well for our purposes. Our API consists mostly of GET calls that return very complex objects (lots of nested models) so for us having it displayed at the top, below the implementation notes, is pretty critical -- I think people get a lot more benefit out of having it there than if it were _only_ somewhere at the bottom, hidden among the response messages.

Anyway, back to the inconsistencies. Here's what's happening in v2.1.2-M2:

**Exhibit A: It's still at the top, but not at the bottom.**

![screen shot 2015-05-11 at 1 11 48 pm](https://cloud.githubusercontent.com/assets/855115/7570614/65376a54-f7df-11e4-87ab-a1850fd2eb5a.png)
![screen shot 2015-05-11 at 1 12 02 pm](https://cloud.githubusercontent.com/assets/855115/7570616/6950b0a0-f7df-11e4-8d41-34979dcd4e50.png)

(and yes, as a side note, the heading at the top is wrong, as there is no response code 200 for this call -- I think I previously pointed that out in #1147)

**Exhibit B: It's not at the top, only at the bottom (under `default`).**

![screen shot 2015-05-11 at 1 14 36 pm](https://cloud.githubusercontent.com/assets/855115/7570651/c8460fba-f7df-11e4-91da-4effe139943b.png)
![screen shot 2015-05-11 at 1 14 49 pm](https://cloud.githubusercontent.com/assets/855115/7570654/cb68483e-f7df-11e4-8334-1f757be27aef.png)

It seems like the inconsistency is related to whether the operation has a possible response code of 200 or not. It also seems like it's _only_ listening to the `type` parameter on the operation (as opposed to the `responseModel` property on the responseMessage) because I have a one-off example that doesn't follow the consistent pattern of all our other operations:

``` javascript
        "path": "/identity/v2/stations",
            "description": "Set a user's favorite stations",
            "operations": [
                {
                    "method": "PUT",
                    "summary": "Update the logged-in user's favorite station(s)",
                    "nickname": "updateStations",
                    // NOTE: no "type" here!!!
                    "parameters": [
                        ...
                    ],
                    "responseMessages": [
                        {
                            "code": 201,
                            "message": "Successfully updated the user primary station or there was no change",
                            "responseModel": "UserDocument" // <----------------
                        },
```

which outputs like:

![screen shot 2015-05-11 at 1 23 19 pm](https://cloud.githubusercontent.com/assets/855115/7570809/f6889c5c-f7e0-11e4-89e7-209ca5e96ba4.png)
![screen shot 2015-05-11 at 1 23 31 pm](https://cloud.githubusercontent.com/assets/855115/7570811/fa0fb176-f7e0-11e4-8478-2d23426ee9d1.png)

-- in other words, no model anywhere. This would suggest that the `responseModel` field on responseMessages is just being ignored altogether (which I'm guessing is the same bug as #1264 -- so it's not just error responses).

**Comment by webron** [comment]
Some of the issues you mention here are real, some are limitations (can't be fixed) and some are not.

What we do in the current version of the UI is migration 1.2 definitions to 2.0. As such, we have to deal with some lack of information. Since there was no way to define the code of a successful response, we are left to assume it's 200. So whether you declare 201 or not is irrelevant, 200 will be added automatically. This is a limitation and I don't see it being changed.

The case where you have a 200 response declared and its model not being defined, that's indeed a bug that should be resolved. The challenge would be if the operation type and the 200 responseModel are not the same, but in any case we should just pick one and go with it. There's no other way to solve it.

The last case where there's no `type` - that's a problem with your definition. It's simply not a valid Swagger 1.2 definition as `type` for operation is mandatory.

**Comment by webron** [comment]
@xiehan - need your help here a bit. Some of the issues here were also raised in #1266 if I'm not mistaken, and the others may not be an issue as explained above. Can you please go over it so we'll see what's left?

**Comment by webron** [comment]
@xiehan - will you have time to go over it?

**Comment by webron** [comment]
3.0 has changed the way responses are rendered.
